trigger/virtcontainers/vc.go

```diff
@@ -67,10 +121,32 @@ func monitorPods(vcRunRoot, throttler string) error {
 	}
 	defer watcher.Close()
 
-	wg.Add(1)
+	// Wait for vc root if it does not exist
+	if _, err := os.Stat(vcRunRoot); os.IsNotExist(err) {
+		if err := waitForDirectory(vcRunRoot); err != nil {
+			logrus.Errorf("Could not monitor virtcontainers base directory %s: %v", vcRunRoot, err)
+			return err
+		}
+	}
+
+	// Wait for the pods path if it does not exist
+	podsPath := filepath.Join(vcRunRoot, "pods")
+	if _, err := os.Stat(podsPath); os.IsNotExist(err) {
+		if err := waitForDirectory(podsPath); err != nil {
+			logrus.Errorf("Could not monitor virtcontainers pods %s: %v", podsPath, err)
+			return err
+		}
 
-	logrus.Debugf("Monitoring virtcontainers event at %s", vcRunRoot)
+		// First pod created, we should kick the throttler
```

Comment:
I think this could trigger before vc has actually created the pod-specific directory and launched the vm, etc. Would it be safer to look for a subdir below `pods/` that contains `state.json` or one of the other state files?

Reply:
My assumption here is that we hit that code path when the `pods` directory did not exist. If we wait for it and it shows up, this means that at least `virtcontainers` is creating a new pod (It won't create `pods` and leave it empty). So it's very likely that it will start a new pod as well. Does that make sense ?